modules/data/published_post.py

```diff
@@ -0,0 +1,112 @@
+
+    @classmethod
+    def from_channel(cls, channel_id: int, c_message_id: int):
+        """Retrieves a published post from the info related to the channel
+
+        Args:
+            channel_id (int): id of the channel
+            c_message_id (int): id of the post in the channel
+
+        Returns:
+            PublishedPost: istance of the class
+        """
+        is_present = DbManager.count_from(table_name="published_meme",
+                                          where="channel_id = %s and c_message_id = %s",
+                                          where_args=(channel_id, c_message_id))
+        if not is_present:
+            return None
+
+        return cls(channel_id=channel_id, c_message_id=c_message_id)
+
+    def __get_user_vote(self, user_id: int) -> Optional[str]:
+        """Gets the vote of a specific user on the post
+
+        Args:
+            user_id (int): id of the user that voted
+
+        Returns:
+            Optional[str]: value of the vote or None if a vote was not yet made
+        """
+        vote = DbManager.select_from(select="vote",
+                                     table_name="votes",
+                                     where="user_id = %s and c_message_id = %s and channel_id = %s",
+                                     where_args=(user_id, self.c_message_id, self.channel_id))
+
+        if len(vote) == 0:  # the vote is not present
+            return None
+        return vote[0]['vote']
+
+    def set_user_vote(self, user_id: int, vote: str) -> bool:
+        """Adds the vote of the user on the post, updates the existing vote or deletes it
+
+        Args:
+            user_id (int): id of the user that voted
+            vote (str): kind of vote the user wants to add or remove
+
+        Returns:
+            bool: whether the vote was added or removed
+        """
+        current_vote = self.__get_user_vote(user_id=user_id)
+        vote_added = True
+        if current_vote is None:  # there isn't a vote yet
+            DbManager.insert_into(table_name="votes",
+                                  columns=("user_id", "c_message_id", "channel_id", "vote"),
+                                  values=(user_id, self.c_message_id, self.channel_id, vote))
+        elif current_vote != vote:  # the old vote was different from the new vote
+            DbManager.query_from_string(f"UPDATE votes SET vote = {vote}\
+                                        WHERE user_id = '{user_id}'\
+                                        and c_message_id = '{self.c_message_id}'\
+                                        and channel_id = '{self.channel_id}'")
+        else:  # the user wants to remove his vote
+            DbManager.delete_from(table_name="votes",
+                                  where="user_id = %s and c_message_id = %s and channel_id = %s",
+                                  where_args=(user_id, self.c_message_id, self.channel_id))
+            vote_added = False
```

Comment:
replace this with `return False` and put `return True` at the end, you can remove vote_added